### PR TITLE
unnecessary iostream removal in cuda

### PR DESCRIPTION
**Pull request by dhollman**
Most `#include <iostream>` in Cuda appears to be leftovers from people using prints for debugging.

### CODE REVIEW

File: core/src/Cuda/Kokkos_Cuda_BlockSize_Deduction.hpp [hunk 1]
```diff
@@ -121,10 +116,6 @@ int cuda_get_max_block_size(const CudaInternal* cuda_instance,
         opt_threads_per_sm = threads_per_sm;
       }
     }
-    // printf("BlockSizeMax: %i Shmem: %i %i %i %i Regs: %i %i Blocks: %i %i
-    // Achieved: %i %i Opt: %i %i\n",block_size,
-    //   shmem_per_sm,max_shmem_per_block,functor_shmem,total_shmem,
-    //   regs_per_sm,regs_per_thread,max_blocks_shmem,max_blocks_regs,blocks_per_sm,threads_per_sm,opt_block_size,opt_threads_per_sm);
```
Comment (dalg24) [inline core/src/Cuda/Kokkos_Cuda_BlockSize_Deduction.hpp:127]: For one these print statement are useful to debug, but also they will conflict with #3103

File: core/src/impl/Kokkos_Error.cpp [hunk 2]
```diff
@@ -51,6 +51,7 @@
 #include <iomanip>
 #include <stdexcept>
 #include <impl/Kokkos_Error.hpp>
```
Comment (dalg24) [inline core/src/impl/Kokkos_Error.cpp:53]: Remove that include

File: core/src/Cuda/Kokkos_Cuda_Parallel.hpp [hunk 3]
```diff
@@ -264,379 +263,7 @@ class TeamPolicyInternal<Kokkos::Cuda, Properties...>
...
-          min(static_cast<index_type>(m_rp.m_tile_end[2] * m_rp.m_tile_end[3]),
-              static_cast<index_type>(maxblocks)),
-          min(static_cast<index_type>(m_rp.m_tile_end[4] * m_rp.m_tile_end[5]),
-              static_cast<index_type>(maxblocks)));
-      CudaParallelLaunch<ParallelFor, LaunchBounds>(
-          *this, grid, block, 0, m_rp.space().impl_internal_space_instance(),
-          false);
-    } else {
-      printf("Kokkos::MDRange Error: Exceeded rank bounds with Cuda\n");
+  printf("Kokkos::MDRange Error: Exceeded rank bounds with Cuda\n");
```
Comment (dalg24) [inline core/src/Cuda/Kokkos_Cuda_Parallel.hpp:266]: Did you mean to change this?

File: core/src/Cuda/Kokkos_Cuda_BlockSize_Deduction.hpp (hunk 1 above)
Comment (dhollman) [reply core/src/Cuda/Kokkos_Cuda_BlockSize_Deduction.hpp:127]: If they're useful for debugging, they shouldn't be commented out, but surrounded with `#ifdef KOKKOS_DEBUG` or something like that (or `#ifdef KOKKOS_IMPL_CUDA_BLOCK_SIZE_DEBUG` if that is preferable). Letting code just rot behind a comment isn't helpful, though.

File: core/src/Cuda/Kokkos_Cuda_Parallel.hpp (hunk 3 above)
Comment (dhollman) [reply core/src/Cuda/Kokkos_Cuda_Parallel.hpp:266]: umm... no... how did that happen?

File: core/src/impl/Kokkos_Error.cpp (hunk 2 above)
Comment (dhollman) [reply core/src/impl/Kokkos_Error.cpp:53]: this is a `cpp` file, so it's not that important, but I'm not sure which include you're referring to.

Comment (dalg24) [reply core/src/impl/Kokkos_Error.cpp:53]: `<impl/Kokkos_Error.hpp>`

File: core/src/Cuda/Kokkos_Cuda_BlockSize_Deduction.hpp (hunk 1 above)
Comment (dalg24) [reply core/src/Cuda/Kokkos_Cuda_BlockSize_Deduction.hpp:127]: I agree with everything you said.  I would prefer if you did not touch that file but I can resolve conflicts later.

File: core/src/impl/Kokkos_Error.cpp [hunk 4]
```diff
@@ -156,4 +157,19 @@ void traceback_callstack(std::ostream &msg) {
 }
 
 }  // namespace Impl
+
+#ifdef KOKKOS_ENABLE_CUDA
+namespace Experimental {
+
+void CudaRawMemoryAllocationFailure::append_additional_error_information(
+    std::ostream &o) const override {
```
Comment (masterleinad) [inline core/src/impl/Kokkos_Error.cpp:165]: /var/jenkins/workspace/Kokkos/core/src/impl/Kokkos_Error.cpp(165): error: expected a "{"

File: core/src/Cuda/Kokkos_Cuda_BlockSize_Deduction.hpp (hunk 1 above)
Comment (dhollman) [reply core/src/Cuda/Kokkos_Cuda_BlockSize_Deduction.hpp]: Okay I reverted it

File: core/src/impl/Kokkos_Error.cpp (hunk 2 above)
Comment (dhollman) [reply core/src/impl/Kokkos_Error.cpp:53]: I don't understand. `<impl/Kokkos_Error.hpp>` is used throughout this file (for instance, `Experimental::RawMemoryAllocationFailure` is there).

File: core/src/Cuda/KokkosExp_Cuda_IterateTile_Refactor.hpp [hunk 5]
```diff
@@ -1562,7 +1560,6 @@ struct DeviceIterateTile<
             tile_idx /= m_rp.m_tile_end[i];
 
             // tile-local indices identified with (index_type)threadIdx.y
-            m_local_offset[i] = (thrd_idx % m_rp.m_tile[i]);
```
Comment (masterleinad) [inline core/src/Cuda/KokkosExp_Cuda_IterateTile_Refactor.hpp:1565]: Did you remove this line intentionally?

Comment (dhollman) [reply core/src/Cuda/KokkosExp_Cuda_IterateTile_Refactor.hpp:1565]: no. What?? That wasn't in the `git commit -p`...

File: core/src/impl/Kokkos_Error.cpp (hunk 2 above)
Comment (dalg24) [reply core/src/impl/Kokkos_Error.cpp:53]: https://github.com/kokkos/kokkos/blob/5b791388dcd3abcd199b2af76c9a0f440cd956f3/core/src/Cuda/Kokkos_Cuda_Error.hpp#L51

**Comment by dalg24** [comment]
Clean the history before we merge

**Comment by masterleinad** [comment]
You will also need to run `clang-format` again. :wink:

**Pull request merged**